cypress-E2E-tests/cypress/integration/asset_properties_widget.cy.ts

```diff
@@ -0,0 +1,614 @@
+    const keyElement = ':nth-child(4) > :nth-child(4) > span';
+    cy.selectAsset(assetName);
+    cy.get(asset_properties_widget_elements.addPropertyButton).click();
+    cy.selectAssetProperty(title);
+    cy.get(asset_properties_widget_elements.selectButton).click();
+    cy.get(keyElement).should('have.text', title);
+    cy.get(asset_properties_widget_elements.saveButton).click();
+    cy.get(cardTitleElement).should('contain.text', assetProperties);
+    cy.deleteCard();
+  });
+
+  // Verify the presence of heading Asset Properties and description text
+  it('TC_Asset_Properties_Widget_config_019', () => {
+    const description = 'Editable form for asset properties';
+    cy.contains(assetProperties);
+    cy.contains(description);
+  });
+
+  // Verify the presence of text asset selection and search bar with message"search for group or asset".
+  it('TC_Asset_Properties_Widget_config_020', () => {
+    const selectionTextElement = '.p-b-8 > .text-medium';
+    cy.get(selectionTextElement).should('have.text', ' Asset selection ');
+    cy.get(searchElement).should('exist');
+    cy.get(assetsTextElement).should('contain.text', 'Assets');
+  });
+
+  // User should be able to select the asset.
+  it('TC_Asset_Properties_Widget_config_021', () => {
+    cy.selectAsset(assetName);
+    cy.get(asset_properties_widget_elements.saveButton).should('be.enabled');
+    cy.get(asset_properties_widget_elements.cancelButton).should('be.enabled');
+  });
+
+  // Verify that on click of an asset in Asset selection section it should navigate to the next screen displaying its subassets
+  it('TC_Asset_Properties_Widget_config_022', () => {
+    const assetName1 = 'check 1';
+    const subassetName = 'check 2';
+    cy.get(asset_properties_widget_elements.loadMoreButton).click();
+    cy.get(`p[title='${assetName1}']`).click();
+    cy.get(textElement).should('contains.text', 'Assets > ' + assetName1);
+    cy.get(`p[title='${subassetName}']`).should('exist');
+  });
+
+  // Also user can navigate back  to respective parent asset using '<'.
+  it('TC_Asset_Properties_Widget_config_023', () => {
+    const assetName1 = 'check 1';
+    const subassetName = 'check 2';
+    cy.get(asset_properties_widget_elements.loadMoreButton).click();
+    cy.get(`p[title='${assetName1}']`).click();
+    cy.get(textElement).should('contains.text', 'Assets > ' + assetName1);
+    cy.get(`p[title='${subassetName}']`).should('exist');
+    cy.get(backArrowElement).click();
+    cy.get(assetsTextElement).should('contain.text', 'Assets');
+  });
+
+  // User can select only one asset at a time.
+  it('TC_Asset_Properties_Widget_config_024', () => {
+    const assetName = 'check 1';
+    const assetName1 = 'check 3';
+    cy.get(asset_properties_widget_elements.loadMoreButton).click();
+    cy.selectAsset(assetName);
+    cy.selectAsset(assetName1);
+    cy.get(`div[title='Assets > ${assetName}']`)
+      .children('div[class*="m-l-4 m-r-4 miller-column__item__checkbox"]')
```

Comment:
If you are identifying an element with matching attribute value, mentioning matching text is fine. For ex:
```suggestion
      .children('div[class*="checkbox"]')
```

Reply:
done